### PR TITLE
Update Sovrn bid adapter to send tmax

### PR DESCRIPTION
**Pull request by ikagotso**

### CODE REVIEW

**Comment by smounitz** [comment]
The way I understand prebid timeout:

Prebid.js has a default timeout of 3000ms with a buffer of 400ms. So if a publisher does not explicitly set the timeout in  the config  `pbjs.setConfig({ bidderTimeout: 3000})) `  the bidders will have 2600ms to respond.

Can we please check this,  and also add a test asserting that a default tmax is passed even when the publisher does not set a timeout

File: test/spec/modules/sovrnBidAdapter_spec.js [hunk 1]
```diff
@@ -170,6 +170,19 @@ describe('sovrnBidAdapter', function() {
         expect(payload.site.domain).to.equal('example.com')
       })
 
+      it('sets correct timeout', function() {
+        const bidderRequest = {
+          ...baseBidderRequest,
+          bidderCode: 'sovrn',
+          auctionId: '1d1a030790a475',
+          bidderRequestId: '22edbae2733bf6',
+          timeout: 3000,
```
Comment (smounitz) [inline test/spec/modules/sovrnBidAdapter_spec.js:179]: 3000 is the default so let's test a smaller timeout

**Review by bharris-sovrn — APPROVED** [review]
We should add merge protections to this repo.

**Comment by ikagotso** [comment]
> The way I understand prebid timeout:
> 
> Prebid.js has a default timeout of 3000ms with a buffer of 400ms. So if a publisher does not explicitly set the timeout in the config `pbjs.setConfig({ bidderTimeout: 3000})) ` the bidders will have 2600ms to respond.
> 
> Can we please check this, and also add a test asserting that a default tmax is passed even when the publisher does not set a timeout

According to this page: https://docs.prebid.org/features/timeouts.html

you are talking about Auction Timeout (Default 3000MS) and Timeout Buffer (Default 400MS).

Shouldn't we be taking into consideration Prebid Server s2sConfig Timeout ?

**Comment by smounitz** [comment]
Irakli and I looked into the timeout buffer and it turns out that it gets added to the timeout. 
We also tested that if the publisher doesn't set a timeout the default of 3000ms is passed.

So everything is good to go

**Review by smounitz — APPROVED** [review]

File: test/spec/modules/sovrnBidAdapter_spec.js (hunk 1 above)
Comment (smounitz) [reply test/spec/modules/sovrnBidAdapter_spec.js:179]: after looking into this this test is fine

Comment (smounitz) [reply test/spec/modules/sovrnBidAdapter_spec.js:179]: initially i was concerned that if we test for 3000 we wouldn't know if it came from the publisher timeout setting or the default. But it turns out that this tests are for the module only and don't get the default anyway.

**Comment by ikagotso** [comment]
Not merging because similar PR is on original Prebid.js repo

**Comment by dmitrybor** [comment]
Merged into the original Prebid.js repo and synced back to master

**Pull request closed without merging**